keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +165,155 @@ Running this set of tests is optional for a per-provider basis. We would
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. The `volume.beta.kubernetes.io/migrated-to` annotation will be used by `v1.6.0+` of the CSI External provisioner to determine whether a PV or PVC should be operated on by the provisioner. The annotation will be set (and removed on rollback) for Kubernetes `v1.17.2+`, we will carefully document the fact that rollback with migration on may not be successful to a version before `v1.17.2`. The benefit being that PV Controller start-up annealing of this annotation will allow the PV Controller to stand down and the CSI External Provisioner to pick up a PV that was dynamically provisioned before migration was enabled. These newer external provisioners will still be compatible with older versions of Kubernetes with migration on even if they do not set the migrated-to annotation. However, without the migrated-to annotation a newer provisioner with a Kubernetes cluster `<1.17.2` will not be able to delete volumes provisioned before migration until the Kubernetes cluster is upgraded to `v1.17.2+`.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+The CSI migration will continue to function.
+
+* **Are there any tests for feature enablement/disablement?**
+We have CSI Migration test for each plugin that are implemented and maintained by each driver maintainer.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  The rollout can fail if the ordering of CSIMigration{cloud-provider} flag was wrongly enabled on kubelet and kube-controller-manager. Specifically, if on the node side kubelet enables the flag and control-plane side the flag is not enabled, then the volume will not be able to be mounted successfully.
+
+* **What specific metrics should inform a rollback?**
+  We have metrics on the CSI sidecar side called `csi_operation_duration_seconds` and core k8s metrics on both kube-controller-manager and kubelet side called `storage_operation_duration_seconds`. Both of them will have a `migrated` field to whether indicate this operation is a migrated PV operation. If the error ratio of such PV has an unusual strike or have an overall high error ratio, it means something goes wrong and we need to rollback.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+To turn it on by default in Beta, we require each in-tree plugin to at least manually test the upgrade->downgrade->upgrade path.
+For GA, we require such test exists in each driver's test CI.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+There will not be API removal in CSI migration itself. But eventually when CSI migration is all finished. We will plan to remove all in-tree plugins. So deprecation and code-removal will be required eventually.
```

Comment:
As each plugin goes to beta, we also deprecate the intree-plugin for it.

Some CSI drivers are not able to maintain 100% backwards compatibility, so those drivers need to deprecate those behaviors.